### DISCUSSION TITLE
How do I get uvicorn to forcefully shutdown, ignoring connections / background tasks / etc.?

### DISCUSSION THREAD

**Discussion by vedantroy** [Kludex/uvicorn#2100] [Q&A] [answered]
I need to kill uvicorn from within my fast api application. I'm trying to use pkill, but there's a problem.
- When I send SIGKILL, the socket becomes unavailable when restarting the server
- If I send SIGINT, uvicorn will wait for background tasks, etc. to finish -- prompting me w/ "Press Ctrl-C" to force-quit. Sending SIGINT twice in a row doesn't seem to work either.

What should I be doing?

**Comment by Kludex** [comment, marked as answer]
Sending two SIGTERM or SIGINT should work. 🤔

**Reply by vedantroy** [reply]
Doing:
```
import time
def launch_uvicorn():
    return subprocess.Popen('uvicorn main:app --host 0.0.0.0 --port 10000', shell=True)

proc = launch_uvicorn()
time.sleep(10)
proc.send_signal(signal.SIGINT)
proc.send_signal(signal.SIGINT)
while True:
    proc.send_signal(signal.SIGINT)
    proc.send_signal(signal.SIGINT)
    if proc.returncode is None:
        print(f"Waiting for uvicorn to terminate")
        time.sleep(5)
    else: break
```

Does not seem to work.

**Reply by vedantroy** [reply]
Whoops, was not using proc.poll
